### PR TITLE
chore: setup

### PR DESCRIPTION
**Pull request by nejcm**
- added testing
- added ci & semantic release
- added prettier
- updated packages
- added react-query
- fixed some lint issues

Mostly basic setup updates.

### CODE REVIEW

**Comment by nejcm** [comment]
This is the first CI run here: https://github.com/pendulum-chain/portal/actions/runs/4229253424/jobs/7345462209

Will check why build is failing.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      #- name: 🧪 Cypress
+      #  uses: cypress-io/github-action@v5.0.0
+      #  with:
+      #    install: true
+      #    install-command: yarn cypress install --force
+      #    start: yarn start
+      #    working-directory: packages/web-app
+      #    config-file: cypress.ci.config.js
+      #    wait-on: 'http://localhost:3000'
+      #    wait-on-timeout: 300
```
[the remark this answers is not in the document]
Comment (nejcm) [reply .github/workflows/ci.yml:72]: Yeah. Wanted to add cypress but will leave it out until you decide if you want it.

**Comment by nejcm** [comment]
Also I added prettier but will leave it up to you regarding what are config options that you guys use. Just added some base one there.

**Comment by ebma** [comment]
> Regarding node and the failing build. It needs to be updated in netlify as I am seeing it's failing there.

I set the yarn and node version now and the CI job passes but netlify won't show me the logs of the automatic deployment at the moment. I'll check back on this tomorrow.
```
4:52:16 PM: [ERROR] Deploy logs are currently unavailable. We are working on resolving the issue.
```

**Comment by ebma** [comment]
I resolved the deploy issue now by setting `YARN_ENABLE_IMMUTABLE_INSTALLS=false` for netlify deployments.

**Pull request merged**